server/service/client_mdm.go

```diff
@@ -146,8 +148,32 @@ func (c *Client) ValidateBootstrapPackageFromURL(url string) (*fleet.MDMAppleBoo
 	return downloadRemoteMacosBootstrapPackage(url)
 }
 
-func downloadRemoteMacosBootstrapPackage(url string) (*fleet.MDMAppleBootstrapPackage, error) {
-	resp, err := http.Get(url) // nolint:gosec // we want this URL to be provided by the user. It will run on their machine.
+func extractFilenameFromPath(p string) string {
+	u, err := url.Parse(p)
+	if err != nil {
+		return ""
+	}
+
+	invalid := map[string]struct{}{
+		"":  {},
+		".": {},
+		"/": {},
+	}
```

Comment:
thank you! so, this list is not being used to check if the file _contains_ the character, those are the current return values of `path.Base` and `path.Ext` for some edge cases.

The check you suggest sounds like a good idea, should we do it upstream? I'm thinking in the endpoint handler. What do you think?

Reply:
Oh right! So in the endpoint handler, that would be where it receives whatever filename we ended with, after extraction? That would be the best place, yes, in the service layer - so regardless of whether the filename was user-provided or determined from the URL, it would get through that validation.